conans/model/build_info.py

```diff
@@ -302,31 +313,67 @@ def __getattr__(self, item):
         return attr
 
     @staticmethod
-    def _merge_lists(seq1, seq2):
-        return seq1 + [s for s in seq2 if s not in seq1]
+    def _merge_lists(seq1, seq2, reverse=False):
+        if reverse:
+            return seq2 + [s for s in seq1 if s not in seq2]
+        else:
+            return seq1 + [s for s in seq2 if s not in seq1]
 
-    def _aggregated_values(self, item):
+    def _aggregated_values(self, item, reverse=False):
         values = getattr(self, "_%s" % item)
         if values is not None:
             return values
         values = getattr(self._cpp_info, item)
         if self._cpp_info.components:
-            for component in self._cpp_info.components.values():
-                values = self._merge_lists(values, getattr(component, item))
+            for component in self._get_sorted_components().values():
+                values = self._merge_lists(values, getattr(component, item), reverse)
         setattr(self, "_%s" % item, values)
         return values
 
-    def _aggregated_paths(self, item):
+    def _aggregated_paths(self, item, reverse=False):
         paths = getattr(self, "_%s_paths" % item)
         if paths is not None:
             return paths
         paths = getattr(self._cpp_info, "%s_paths" % item)
         if self._cpp_info.components:
-            for component in self._cpp_info.components.values():
-                paths = self._merge_lists(paths, getattr(component, "%s_paths" % item))
+            for component in self._get_sorted_components().values():
+                paths = self._merge_lists(paths, getattr(component, "%s_paths" % item), reverse)
         setattr(self, "_%s_paths" % item, paths)
         return paths
 
+    def _get_sorted_components(self):
```

Comment:
Should this be a public method or a property so people are able to access this functionality in generators?

Reply:
No. IMO we should write a `BaseGenerator` class providing this layer of abstraction on top of `deps_cpp_info` and implementing some logic that is scattered (and duplicated) in many generators.

Reply:
Agree